src/mlpack/tests/cyclegan_test.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file cyclegan_test.cpp
+ * @author Shikhar Jaiswal
+ *
+ * Tests the CycleGAN network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/loss_functions/cross_entropy_error.hpp>
+#include <mlpack/methods/ann/gan/gan.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/softmax_regression/softmax_regression.hpp>
+#include <mlpack/core/optimizers/adam/adam.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::math;
+using namespace mlpack::optimization;
+using namespace mlpack::regression;
+using namespace std::placeholders;
+
+BOOST_AUTO_TEST_SUITE(CycleGANNetworkTest);
+
+/*
+ * Tests the CycleGAN implementation on MNIST to SVHN conversion.
+ * It's not viable to train on bigger parameters due to time constraints.
+ * Please refer mlpack/models repository for the tutorial.
+ *
+BOOST_AUTO_TEST_CASE(CycleGANMNISTToSVHNTest)
```

Comment:
Do you think we could add a really simple test for the Cycle GAN?

Reply:
I will have to search up a bit, but as far as I have seen, this really is the simplest and least resource intensive test that I have encountered.

Reply:
Do you have the travis timings for the test? Perhaps we can comment this one in?

Reply:
I'm yet to test this out on the full dataset, I got busy with formulating the ideas for the paper, so this had to wait.

Reply:
Sure, no worries.